backend/src/modules/doubles-tournament/core/application/event/EnqueueMatchWhenTournamentWasStarted.ts

```diff
@@ -0,0 +1,33 @@
+import {EventHandler} from "../../../../../shared/core/application/event/EventHandler";
+import {CommandPublisher} from "../../../../../shared/core/application/command/CommandBus";
+import {EnqueueMatch} from "../command/EnqueueMatch";
+import {TournamentTreeRepository} from "../../../../tournament-tree/core/application/TournamentTreeRepository";
+
+export class EnqueueMatchWhenTournamentWasStarted implements EventHandler<TournamentWasStarted> {
+    constructor(private readonly commandPublisher: CommandPublisher, private readonly repository: TournamentTreeRepository) {}
+
+    /** Nie wiem, które rozwiązanie lepsze */
+
+    async handle(event: TournamentWasStarted): Promise<void> {
+        const tournamentTree = await this.repository
+            .findByTournamentTreeId(event.tournamentId)
+            // .then((tournamentTree) => tournamentTree?.getMatchesQueueReadyToBegin().forEach((matchToEnqueue) => {
+            //     this.commandPublisher.execute(new EnqueueMatch({
+            //         tournamentId: event.tournamentId,
+            //         matchNumber: matchToEnqueue.matchNumberInSequence,
+            //         team1Id: matchToEnqueue.firstTeam?.teamId.raw,
+            //         team2Id: matchToEnqueue.secondTeam?.teamId.raw
+            //     }));
+            // }));
+
+        const matchesToEnqueue = tournamentTree?.getMatchesQueueReadyToBegin()
+            .forEach((matchToEnqueue) => {
+                this.commandPublisher.execute(new EnqueueMatch({
+                    tournamentId: event.tournamentId,
+                    matchNumber: matchToEnqueue.matchNumberInSequence as number,
+                    team1Id: matchToEnqueue.firstTeam?.teamId.raw as string,
+                    team2Id: matchToEnqueue.secondTeam?.teamId.raw as string
```

Comment:
ja byłbym na za drugim :-) 
cały czas używamy async /await, więc chyba fajniej trzymać się tej konwencji 

to as number, as string, as string w liniach 27-29 chyba niepotrzebne - u mnie nie rzucają błędu, a przecież matchNumber i tak przyjdzie jako number z FightingTeamsGroup, a z reszty wyciągasz jako raw

Reply:
no właśnie musiałem dać 'as' bo krzyczy, że typy nie pasują - firstTeam w drzewku ma typ string | undefined, a team1Id w EnqueueMatch ma tylko string, chyba że coś przeoczyłem :P

Reply:
Proponuje zrobić, że `getMatchesQueueReadyToBegin` z drzewka zwraca typ np. `MatchRedyToStart` zmieniłbym nazwe metody z Start, skoro mamy StartMatch command i a nie Beginmatch. I wtedy mapujemy to co znajdziemy juz w drzewku na typ `MatchRedyToStart` a on miałby te pola jako nie undefined.